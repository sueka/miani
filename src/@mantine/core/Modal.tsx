import { useDraggable, useDroppable } from '@dnd-kit/core'
import { Modal as OrigModal, type ModalProps } from '@mantine/core'
import { useLayoutEffect, useMemo, useState } from 'react'
import { tss } from 'tss-react'
import { v4 } from 'uuid'
import assert from '../../lib/assert'

interface Inset {
  top: number
  left: number
}

const useDraggableStyles = tss
  .withParams<{
    dragging: Inset | null
  }>()
  .create(({ dragging }) => ({
    content:
      dragging !== null
        ? {
            position: 'relative',
            top: `${dragging.top}px`,
            left: `${dragging.left}px`,
          }
        : {},
  }))

const Modal: React.FC<ModalProps> = ({
  opened,
  onClose,
  title,
  children,
  ...props
}) => {
  const id = useMemo(v4, [])
  const [dragging, setDragging] = useState<Inset | null>(null)

  const {
    setNodeRef: draggable,
    setActivatorNodeRef: handler,
    active,
    listeners,
    attributes,
  } = useDraggable({
    id,
  })

  const { setNodeRef: droppable } = useDroppable({
    id,
  })

  const { classes: draggableClasses } = useDraggableStyles({
    dragging,
  })

  useLayoutEffect(() => {
    // dragging
    if (active !== null) {
      const { initial, translated } = active.rect.current

      assert(initial !== null)
      assert(translated !== null)

      if (active.id === id) {
        setDragging({
          top: translated.top - initial.top,
          left: translated.left - initial.left,
        })
      }
    }
  }, [active, active?.rect.current])

  return (
    <OrigModal.Root
      opened={opened}
      onClose={onClose}
      ref={droppable}
      {...props}
    >
      <OrigModal.Overlay />
      <OrigModal.Content
        ref={draggable}
        classNames={draggableClasses}
        {...attributes}
      >
        <OrigModal.Header ref={handler} {...listeners}>
          <OrigModal.Title>{title}</OrigModal.Title>
          <OrigModal.CloseButton data-no-dnd />
        </OrigModal.Header>
        <OrigModal.Body>{children}</OrigModal.Body>
      </OrigModal.Content>
    </OrigModal.Root>
  )
}

export default Modal
