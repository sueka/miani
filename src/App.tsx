import { Container } from '@mantine/core'

import VCardForm from './VCardForm'

const App = () => (
  <Container>
    <VCardForm />
  </Container>
)

export default App
