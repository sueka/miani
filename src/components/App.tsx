import { Container, Group, Stack } from '@mantine/core'
import cls from 'classnames'

import classes from './App.module.css'
import VCardForm from './VCardForm/VCardForm'
import VCardQrCode from './VCardQrCode'
import VCardTextarea from './VCardTextarea/VCardTextarea'

const App = () => (
  <Container>
    <Group align="start">
      <Stack classNames={{ root: classes['FormStack'] }}>
        <VCardForm />
      </Stack>
      <Stack
        classNames={{
          root: cls(classes['VCardStack'], classes['Sticky']),
        }}
      >
        <VCardTextarea />
        <VCardQrCode />
      </Stack>
    </Group>
  </Container>
)

export default App
