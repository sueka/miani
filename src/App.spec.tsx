import { expect, test } from 'bun:test'
import { MantineProvider } from '@mantine/core'
import { render } from 'react-dom'
import { RecoilRoot } from 'recoil'

import App from './App'

test('<App>', () => {
  render(
    <MantineProvider>
      <RecoilRoot>
        <App />
      </RecoilRoot>
    </MantineProvider>,
    document.body,
  )

  expect(document.body.innerHTML).toMatchSnapshot()
})
