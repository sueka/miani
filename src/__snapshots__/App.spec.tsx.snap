// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> 1`] = `"<style data-mantine-styles="classes">@media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}</style><div class="m-7485cace mantine-Container-root"><style data-mantine-styles="inline">.__m__-P0-1{--grid-gutter:var(--mantine-spacing-md);}</style><div class="m-410352e9 mantine-Grid-root __m__-P0-1"><div class="m-dee7bd2f mantine-Grid-inner"><style data-mantine-styles="inline">.__m__-P0-3{--col-flex-grow:auto;--col-flex-basis:100%;--col-max-width:100%;}@media(min-width: 36em){.__m__-P0-3{--col-flex-grow:auto;--col-flex-basis:100%;--col-max-width:100%;}}@media(min-width: 48em){.__m__-P0-3{--col-flex-grow:auto;--col-flex-basis:50%;--col-max-width:50%;}}@media(min-width: 62em){.__m__-P0-3{--col-flex-grow:auto;--col-flex-basis:33.333333333333336%;--col-max-width:33.333333333333336%;}}</style><div class="m-96bdd299 mantine-Grid-col __m__-P0-3"><div style="--stack-gap: var(--mantine-spacing-xs); --stack-align: stretch; --stack-justify: flex-start;" class="m-6d731127 mantine-Stack-root"><div class="m-46b77525 mantine-InputWrapper-root mantine-TextInput-root"><label data-required="true" for="mantine-P0-6" id="mantine-P0-6-label" class="m-8fdc1311 mantine-InputWrapper-label mantine-TextInput-label">FN<span aria-hidden="true" class="m-78a94662 mantine-InputWrapper-required mantine-TextInput-required"> *</span></label><p id="mantine-P0-6-description" class="m-fe47ce59 mantine-InputWrapper-description mantine-TextInput-description">Specifies the <mark>formatted</mark> text corresponding to the <mark>name</mark> of the object the vCard represents.</p><div style="--input-margin-top: calc(var(--mantine-spacing-xs) / 2);" data-variant="default" class="m-6c018570 mantine-Input-wrapper mantine-TextInput-wrapper"><input data-variant="default" placeholder="Mr. John Q. Public, Esq." required="" aria-invalid="false" aria-describedby="mantine-P0-6-description" id="mantine-P0-6" class="m-8fb7ebe7 mantine-Input-input mantine-TextInput-input"></div></div><div class="m-46b77525 mantine-InputWrapper-root mantine-TextInput-root"><label data-required="true" for="mantine-P0-12" id="mantine-P0-12-label" class="m-8fdc1311 mantine-InputWrapper-label mantine-TextInput-label">N<span aria-hidden="true" class="m-78a94662 mantine-InputWrapper-required mantine-TextInput-required"> *</span></label><p id="mantine-P0-12-description" class="m-fe47ce59 mantine-InputWrapper-description mantine-TextInput-description">Specifies the components of the <mark>name</mark> of the object the vCard represents.</p><div style="--input-margin-top: calc(var(--mantine-spacing-xs) / 2);" data-variant="default" class="m-6c018570 mantine-Input-wrapper mantine-TextInput-wrapper"><input data-variant="default" placeholder="Public;John;Quinlan;Mr.;Esq." required="" aria-invalid="false" aria-describedby="mantine-P0-12-description" id="mantine-P0-12" class="m-8fb7ebe7 mantine-Input-input mantine-TextInput-input"></div></div><div class="m-46b77525 mantine-InputWrapper-root"><label for="mantine-P0-18" id="mantine-P0-18-label" class="m-8fdc1311 mantine-InputWrapper-label">BDAY</label><p id="mantine-P0-18-description" class="m-fe47ce59 mantine-InputWrapper-description">Specifies the <mark>birth date</mark> of the object the vCard represents.</p><div style="--group-gap: var(--mantine-spacing-sm); --group-align: center; --group-justify: flex-start; --group-wrap: wrap;" class="m-4081bf90 mantine-Group-root"><div style="--checkbox-color: var(--mantine-color-blue-filled);" data-label-position="right" data-checked="true" class="m-bf2d988c mantine-Checkbox-root m-5f75b09e mantine-Checkbox-root"><div class="m-5f6e695e mantine-Checkbox-body"><div data-label-position="right" class="m-26062bec mantine-Checkbox-inner"><input id="mantine-P0-23" type="checkbox" class="mantine-focus-auto m-26063560 mantine-Checkbox-input"><svg viewBox="0 0 10 7" fill="none" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" class="m-bf295423 mantine-Checkbox-icon"><path d="M4 4.586L1.707 2.293A1 1 0 1 0 .293 3.707l3 3a.997.997 0 0 0 1.414 0l5-5A1 1 0 1 0 8.293.293L4 4.586z" fill="currentColor" fill-rule="evenodd" clip-rule="evenodd"></path></svg></div><div class="m-d3ea56bb mantine-Checkbox-labelWrapper"></div></div></div><div data-calendar="true"><div class="m-30b26e33 mantine-DatePicker-levelsGroup"><div data-month-level="true"><div class="m-730a79ed mantine-DatePicker-calendarHeader"><button type="button" data-direction="previous" tabindex="0" class="mantine-focus-auto m-2351eeb0 mantine-DatePicker-calendarHeaderControl m-87cf2631 mantine-UnstyledButton-root"><svg viewBox="0 0 15 15" fill="none" xmlns="http://www.w3.org/2000/svg" style="width: 45%; height: 45%; display: block;" data-direction="previous" class="m-367dc749 mantine-DatePicker-calendarHeaderControlIcon"><path d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z" fill="currentColor" fill-rule="evenodd" clip-rule="evenodd"></path></svg></button><button type="button" tabindex="0" class="mantine-focus-auto m-f6645d97 mantine-DatePicker-calendarHeaderLevel m-87cf2631 mantine-UnstyledButton-root">January 2006</button><button type="button" data-direction="next" tabindex="0" class="mantine-focus-auto m-2351eeb0 mantine-DatePicker-calendarHeaderControl m-87cf2631 mantine-UnstyledButton-root"><svg viewBox="0 0 15 15" fill="none" xmlns="http://www.w3.org/2000/svg" style="width: 45%; height: 45%; display: block;" data-direction="next" class="m-367dc749 mantine-DatePicker-calendarHeaderControlIcon"><path d="M3.13523 6.15803C3.3241 5.95657 3.64052 5.94637 3.84197 6.13523L7.5 9.56464L11.158 6.13523C11.3595 5.94637 11.6759 5.95657 11.8648 6.15803C12.0536 6.35949 12.0434 6.67591 11.842 6.86477L7.84197 10.6148C7.64964 10.7951 7.35036 10.7951 7.15803 10.6148L3.15803 6.86477C2.95657 6.67591 2.94637 6.35949 3.13523 6.15803Z" fill="currentColor" fill-rule="evenodd" clip-rule="evenodd"></path></svg></button></div><table class="m-cc9820d3 mantine-DatePicker-month"><thead class="mantine-DatePicker-monthThead"><tr class="mantine-DatePicker-weekdaysRow"><th class="m-18a3eca mantine-DatePicker-weekday">Mo</th><th class="m-18a3eca mantine-DatePicker-weekday">Tu</th><th class="m-18a3eca mantine-DatePicker-weekday">We</th><th class="m-18a3eca mantine-DatePicker-weekday">Th</th><th class="m-18a3eca mantine-DatePicker-weekday">Fr</th><th class="m-18a3eca mantine-DatePicker-weekday">Sa</th><th class="m-18a3eca mantine-DatePicker-weekday">Su</th></tr></thead><tbody class="mantine-DatePicker-monthTbody"><tr class="mantine-DatePicker-monthRow"><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" data-outside="true" aria-label="26 December 2005" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">26</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" data-outside="true" aria-label="27 December 2005" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">27</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" data-outside="true" aria-label="28 December 2005" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">28</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" data-outside="true" aria-label="29 December 2005" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">29</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" data-outside="true" aria-label="30 December 2005" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">30</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" data-outside="true" aria-label="31 December 2005" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">31</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" data-weekend="true" aria-label="1 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">1</button></td></tr><tr class="mantine-DatePicker-monthRow"><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" aria-label="2 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">2</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" data-today="true" aria-label="3 January 2006" tabindex="0" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">3</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" aria-label="4 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">4</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" aria-label="5 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">5</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" aria-label="6 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">6</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" data-weekend="true" aria-label="7 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">7</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" data-weekend="true" aria-label="8 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">8</button></td></tr><tr class="mantine-DatePicker-monthRow"><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" aria-label="9 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">9</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" aria-label="10 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">10</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" aria-label="11 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">11</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" aria-label="12 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">12</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" aria-label="13 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">13</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" data-weekend="true" aria-label="14 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">14</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" data-weekend="true" aria-label="15 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">15</button></td></tr><tr class="mantine-DatePicker-monthRow"><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" aria-label="16 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">16</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" aria-label="17 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">17</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" aria-label="18 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">18</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" aria-label="19 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">19</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" aria-label="20 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">20</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" data-weekend="true" aria-label="21 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">21</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" data-weekend="true" aria-label="22 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">22</button></td></tr><tr class="mantine-DatePicker-monthRow"><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" aria-label="23 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">23</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" aria-label="24 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">24</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" aria-label="25 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">25</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" aria-label="26 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">26</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" aria-label="27 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">27</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" data-weekend="true" aria-label="28 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">28</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" data-weekend="true" aria-label="29 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">29</button></td></tr><tr class="mantine-DatePicker-monthRow"><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" aria-label="30 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">30</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" aria-label="31 January 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">31</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" data-outside="true" aria-label="1 February 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">1</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" data-outside="true" aria-label="2 February 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">2</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" data-outside="true" aria-label="3 February 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">3</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" data-outside="true" aria-label="4 February 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">4</button></td><td data-with-spacing="true" class="m-8f457cd5 mantine-DatePicker-monthCell"><button type="button" data-outside="true" aria-label="5 February 2006" tabindex="-1" class="mantine-focus-auto m-396ce5cb mantine-DatePicker-day m-87cf2631 mantine-UnstyledButton-root">5</button></td></tr></tbody></table></div></div></div></div><div style="--checkbox-color: var(--mantine-color-blue-filled);" data-label-position="right" data-checked="true" class="m-bf2d988c mantine-Checkbox-root m-5f75b09e mantine-Checkbox-root"><div class="m-5f6e695e mantine-Checkbox-body"><div data-label-position="right" class="m-26062bec mantine-Checkbox-inner"><input id="mantine-P0-78" type="checkbox" class="mantine-focus-auto m-26063560 mantine-Checkbox-input"><svg viewBox="0 0 10 7" fill="none" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" class="m-bf295423 mantine-Checkbox-icon"><path d="M4 4.586L1.707 2.293A1 1 0 1 0 .293 3.707l3 3a.997.997 0 0 0 1.414 0l5-5A1 1 0 1 0 8.293.293L4 4.586z" fill="currentColor" fill-rule="evenodd" clip-rule="evenodd"></path></svg></div><div class="m-d3ea56bb mantine-Checkbox-labelWrapper"><label for="mantine-P0-78" class="m-8ee546b8 mantine-Checkbox-label">Omits the year of birth of the object the vCard represents.</label></div></div></div></div><fieldset style="padding-right: var(--mantine-spacing-md); padding-left: var(--mantine-spacing-md); padding-bottom: var(--mantine-spacing-md);" data-variant="default" class="m-84c9523a m-e9408a47 mantine-Fieldset-root"><legend class="m-90794832 mantine-Fieldset-legend">Others</legend><div style="--stack-gap: var(--mantine-spacing-xs); --stack-align: stretch; --stack-justify: flex-start;" class="m-6d731127 mantine-Stack-root"><div class="m-46b77525 mantine-InputWrapper-root mantine-TagsInput-root"><div data-variant="default" data-multiline="true" class="m-6c018570 mantine-Input-wrapper mantine-TagsInput-wrapper"><div data-variant="default" class="m-8fb7ebe7 mantine-Input-input mantine-TagsInput-input"><div class="mantine-TagsInput-pillsList m-1dcfd90b mantine-PillGroup-group"><input data-type="visible" aria-haspopup="listbox" autocomplete="off" id="mantine-P0-84" aria-invalid="false" type="text" class="mantine-TagsInput-inputField m-45c4369d mantine-PillsInputField-field"></div></div></div></div><input type="hidden"></div></fieldset></div></div><style data-mantine-styles="inline">.__m__-P0-93{--col-flex-grow:auto;--col-flex-basis:100%;--col-max-width:100%;}@media(min-width: 36em){.__m__-P0-93{--col-flex-grow:auto;--col-flex-basis:100%;--col-max-width:100%;}}@media(min-width: 48em){.__m__-P0-93{--col-flex-grow:auto;--col-flex-basis:50%;--col-max-width:50%;}}@media(min-width: 62em){.__m__-P0-93{--col-flex-grow:auto;--col-flex-basis:33.333333333333336%;--col-max-width:33.333333333333336%;}}</style><div class="m-96bdd299 mantine-Grid-col __m__-P0-93"><div class="m-46b77525 mantine-InputWrapper-root mantine-Textarea-root"><label for="mantine-P0-95" id="mantine-P0-95-label" class="m-8fdc1311 mantine-InputWrapper-label mantine-Textarea-label">vCard</label><div data-variant="default" data-multiline="true" data-with-right-section="true" class="m-6c018570 mantine-Input-wrapper mantine-Textarea-wrapper"><textarea data-variant="default" readonly="" data-no-overflow="true" aria-invalid="false" id="mantine-P0-95" class="m-8fb7ebe7 mantine-Input-input mantine-Textarea-input"></textarea><div data-position="right" class="m-82577fc2 mantine-Input-section mantine-Textarea-section"><button style="--ai-bg: transparent; --ai-hover: var(--mantine-color-gray-light-hover); --ai-color: var(--mantine-color-gray-light-color); --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;" data-variant="subtle" type="button" class="mantine-focus-auto mantine-active m-8d3f4000 mantine-ActionIcon-root m-87cf2631 mantine-UnstyledButton-root"><span aria-hidden="true" class="m-302b9fb1 mantine-ActionIcon-loader"><span style="--loader-size: calc(var(--ai-size) * 0.55); --loader-color: var(--ai-color);" class="m-b34414df m-5ae2e3c mantine-Loader-root"></span></span><span class="m-8d3afb97 mantine-ActionIcon-icon"><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" style="" class="tabler-icon tabler-icon-copy"><path d="M7 7m0 2.667a2.667 2.667 0 0 1 2.667 -2.667h8.666a2.667 2.667 0 0 1 2.667 2.667v8.666a2.667 2.667 0 0 1 -2.667 2.667h-8.666a2.667 2.667 0 0 1 -2.667 -2.667z"></path><path d="M4.012 16.737a2.005 2.005 0 0 1 -1.012 -1.737v-10c0 -1.1 .9 -2 2 -2h10c.75 0 1.158 .385 1.5 1"></path></svg></span></button></div></div></div></div><style data-mantine-styles="inline">.__m__-P0-108{--col-flex-grow:auto;--col-flex-basis:100%;--col-max-width:100%;}@media(min-width: 36em){.__m__-P0-108{--col-flex-grow:auto;--col-flex-basis:100%;--col-max-width:100%;}}@media(min-width: 48em){.__m__-P0-108{--col-flex-grow:auto;--col-flex-basis:100%;--col-max-width:100%;}}@media(min-width: 62em){.__m__-P0-108{--col-flex-grow:auto;--col-flex-basis:33.333333333333336%;--col-max-width:33.333333333333336%;}}</style><div class="m-96bdd299 mantine-Grid-col __m__-P0-108"><div style="--input-left-section-pointer-events: none; --input-right-section-pointer-events: none;" data-variant="default" class="m-6c018570 mantine-Input-wrapper"><img style="--image-radius: var(--mantine-radius-sm);" class="m-9e117634 mantine-Image-root"></div></div></div></div></div><div data-portal="true"></div><div data-portal="true"></div>"`;
